tuf/ngclient/updater.py

```diff
@@ -122,12 +126,38 @@ def refresh(self) -> None:
             RepositoryError: Metadata failed to verify in some way
             DownloadError: Download of a metadata file failed in some way
         """
+        #                                     json.load()
+        # Load self.spec_version using self._load_local_metadata() here from disk
+        # Exception would be raised if spec_version is not found locally. Set to "1" in that case
+        repository_versions = self._get_repository_versions()
+
+        # Updating self.spec_version
+        specver_message = _get_spec_version(
+            repository_versions, self.spec_version, SUPPORTED_VERSIONS
+        )
+
+        if specver_message(1):
+            logger.warning(specver_message(1))
+        self.spec_version = specver_message(0)
```

Comment:
You use this with an implicit slash in `download_metadata` below, so make sure to add it. Also, if `spec_version` is `None`, we should put an empty string so that the URL formatting works.

```python
self.spec_version = f"{spec_version}/" or ""
```

Reply:
Wouldn't this set `self.spec_version` to `None/` in case `spec_version` turns out to be `None`?
I'm guessing, a better alternative would be :
```
self.spec_version = f"{spec_version}/" if spec_version is not None else ""`
```

Reply:
Good catch! That's totally right

Reply:
Cool, will do that